modules/ctl/src/main/scala/free/interp.scala

```diff
@@ -104,7 +105,7 @@ object interpreter {
   private def doShell(cmd: String \/ List[String], verbose: Boolean, state: IORef[InterpreterState]): EitherT[IO, Int, Output] = {
 
     def handler(s: String): IO[Unit] =
-      if (verbose) doLog(Level.Shell, s, state).run.map(_.toOption.get) // shh
+      if (verbose) doLog(Level.Shell, s, state).run.void
```

Comment:
It was basically an assertion that `doLog` can never fail. I could express this in the type but I think it's ok as-is.